eZ/Publish/Core/REST/Server/Controller/Role.php

```diff
@@ -359,6 +412,47 @@ public function addPolicy($roleId, Request $request)
     }
 
     /**
+     * Adds a policy to a role draft.
+     *
+     * @param $roleId
+     *
+     * @return \eZ\Publish\Core\REST\Server\Values\CreatedPolicy
+     */
+    public function addPolicyByRoleDraft($roleId, Request $request)
+    {
+        $createStruct = $this->inputDispatcher->parse(
+            new Message(
+                array('Content-Type' => $request->headers->get('Content-Type')),
+                $request->getContent()
+            )
+        );
+
+        try {
+            $role = $this->roleService->addPolicyByRoleDraft(
+                $this->roleService->loadRoleDraft($roleId),
+                $createStruct
+            );
+        } catch (LimitationValidationException $e) {
+            throw new BadRequestException($e->getMessage());
+        }
+
+        $policies = $role->getPolicies();
+
+        $policyToReturn = $policies[0];
+        for ($i = 1, $count = count($policies); $i < $count; ++$i) {
+            if ($policies[$i]->id > $policyToReturn->id) {
```

Comment:
@bdunogier It's taken from existing non-draft code by @emodric. The logic is that we want to return the policy we just created, and since the API method doesn't return it, we look for the one with the highest ID.